cpp/oneapi/dal/detail/global_context_impl.hpp

```diff
@@ -0,0 +1,38 @@
+/*******************************************************************************
+* Copyright contributors to the oneDAL project
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/detail/global_context.hpp"
+
+namespace oneapi::dal::detail {
+namespace v1 {
+
+class global_context_impl : public global_context_iface {
+public:
+    global_context_impl();
+
+    const cpu_info_iface &get_cpu_info() const {
+        return *dynamic_cast<const cpu_info_iface *>(&cpu_info_);
```

Comment:
No need for rtti. The inheritance of cpu_info_iface is public in cpu_info, so you can just do:
```
const cpu_info_iface &get_cpu_info() const {
  return cpu_info_;
}
```